### PR TITLE
SDIT-994 Add mapping call for non-association sync

### PR DESCRIPTION
**Pull request by karenmillermoj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/nonassociations/NonAssociationsSynchronisationService.kt [hunk 1]
```diff
@@ -30,27 +44,109 @@ class NonAssociationsSynchronisationService(
...
+      nomisTypeSequence = event.typeSeq,
+      mappingType = NonAssociationMappingDto.MappingType.NOMIS_CREATED,
+    )
+    try {
+      nonAssociationsMappingService.createMapping(
+        mapping,
+        object : ParameterizedTypeReference<DuplicateErrorResponse<NonAssociationMappingDto>>() {},
+      ).also {
+        if (it.isError) {
+          val duplicateErrorDetails = (it.errorResponse!!).moreInfo
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/nonassociations/NonAssociationsSynchronisationService.kt:92]: I still need to add tests for retries/duplicates/failures etc.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/nonassociations/NonAssociationsSynchronisationIntTest.kt [hunk 2]
```diff
@@ -1,7 +1,12 @@
 package uk.gov.justice.digital.hmpps.prisonerfromnomismigration.nonassociations
 
+import com.github.tomakehurst.wiremock.client.WireMock
+import com.github.tomakehurst.wiremock.client.WireMock.anyRequestedFor
```
Comment (karenmillermoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/nonassociations/NonAssociationsSynchronisationIntTest.kt:4]: A lot of changes here - changing the inner classes hasn't helped with the readability, sorry

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/listeners/MigrationMessageType.kt [hunk 3]
```diff
@@ -11,4 +11,5 @@ enum class MigrationMessageType {
 
 enum class SynchronisationMessageType {
   RETRY_SYNCHRONISATION_MAPPING,
+  RETRY_NON_ASSOCIATION_MAPPING,
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/listeners/MigrationMessageType.kt:14]: minor but I think RETRY_SYNCHRONISATION_MAPPING is the exact message you want, so no need to define a new message

**Review by andymarke — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/listeners/MigrationMessageType.kt (hunk 3 above)
Comment (karenmillermoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/listeners/MigrationMessageType.kt:14]: thanks - will remove

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/nonassociations/NonAssociationsMappingService.kt [hunk 4]
```diff
@@ -20,7 +20,7 @@ class NonAssociationsMappingService(@Qualifier("mappingApiWebClient") webClient:
   ): NonAssociationMappingDto? =
     webClient.get()
       .uri(
-        "/mapping/non-associations/first-offender-no/{firstOffenderNo}/second-offender-no/{secondOffenderNo}/type-sequence/{typeSequence}",
```
Comment (karenmillermoj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/nonassociations/NonAssociationsMappingService.kt]: call with additional param not complete yet

**Review by andymarke — APPROVED** [review]
LGTM
